app/src/main/java/com/droidhats/campuscompass/repositories/CalendarRepository.kt

```diff
@@ -0,0 +1,142 @@
+package com.droidhats.campuscompass.repositories
+
+import android.content.ContentUris
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.provider.CalendarContract
+import androidx.lifecycle.MutableLiveData
+import com.droidhats.campuscompass.models.Calendar
+import com.droidhats.campuscompass.models.CalendarEvent
+import java.util.*
+import kotlin.collections.ArrayList
+
+class CalendarRepository {
+
+    private lateinit var userCalendars : MutableMap<String, Calendar>// The user's retrieved calendars
+
+    companion object {
+   // Singleton instantiation
+        private var instance: CalendarRepository? = null
+
+        fun getInstance() =
+            instance
+                ?: synchronized(this) {
+                instance
+                    ?: CalendarRepository().also { instance = it }
+            }
+    }
```

Comment:
This seems to be implementing the thread-safe variant of singleton. 
If this is the case, a disadvantage is that it performs synchronization at every call to CalendarRepository.
We can improve this to make an instance of type enum instead (enum variant of singleton).